### PR TITLE
Added function to support .R files

### PR DESCRIPTION
**Pull request by jeafreezy**
I tested the function using the test.py and it worked fine, I went further by trying to run the signer.py script in the CL but it's not reflecting. None of the files got signed including the java and the .py in the sample folder. Maybe you can try it at your end.

### CODE REVIEW

**Review by Harjacober — COMMENTED** [review]
Hi Emmanuel
So to add support for a new file type, all you just need to do is modify the `symbols py` class and add the extension and it's corresponding comment symbol.

Also as regards the test, once the test runs successfully, be rest assured that you code works fine. The test restores the sample file to it's initial state after signing.

**Comment by jeafreezy** [comment]
Oh okay.

I had to add some supporting functions because .R files doesn't support
block comments.

Jolaiya Emmanuel|Lead YouthMappers FUTA |GIS Analyst|GIS Development and
Data Science Enthusiast.

Department of Remote Sensing and GIS, Federal University of
Technology,Akure.

Email: jolaiyaemmanuel@gmail.com
website: jolaiyaemmanuel.com
contact/WhatsApp: +2348143243024
Twitter: @jeafreezy
LinkedIn & Facebook: Emmanuel Jolaiya

**Comment by Harjacober** [comment]
Oh! Now I understand.
Just thinking. I feel there's a better way we can handle files that doesn't support multi-line comments. It's better we handle them generally. Say we want to support another files that doesn't support multi-line comment like a .sh file, it would be stressful to add additional implementation for it.

So here is an idea I have, why not add an extra field to the symbols key value that specifies the type of comment the extension supports. Say `mul` for multi-line and `one` for single line. So the only function that would need to be changed is the `generateSignature()` function.

**Comment by jeafreezy** [comment]
Sounds great!

I would work on that tomorrow.

Jolaiya Emmanuel|Lead YouthMappers FUTA |GIS Analyst|GIS Development and
Data Science Enthusiast.

Department of Remote Sensing and GIS, Federal University of
Technology,Akure.

Email: jolaiyaemmanuel@gmail.com
website: jolaiyaemmanuel.com
contact/WhatsApp: +2348143243024
Twitter: @jeafreezy
LinkedIn & Facebook: Emmanuel Jolaiya

**Comment by jeafreezy** [comment]
image


It works fine now. Grabbing the first value tuple('#','#') and checking if it's in a list  works. We can add others to the list to check.

File: symbols.py [hunk 1]
```diff
@@ -11,5 +11,6 @@
             "cofee": ('###','###'), 
             "css": ('/*','*/'), 
             "html": ('<!--','-->'),
-            "dart": ('/*','*/')
+            "dart": ('/*','*/'),
+            "R" :('#','#')
```
Comment (Harjacober) [inline symbols.py]: Hi Jeff.
I'm suggesting an extra field be added to the tuple that specifies whether the `symbol` represents a multi-line or single-line comment. For example `symbols = { "R" : ('#','#',False), "Java": ('/*','*/',True) }` the 3rd field in the tuple is `True` for multi-line comment and `False` for single line comments.

File: signer.py [hunk 2]
```diff
@@ -23,8 +23,19 @@ def createdAt(filePath):
...
 # Generates signature content
 def generateSignature(symbol, name, date):
-    signature = "{open}\n**\n * author :   \t{name}\n * created : \t{date}\n**\n{close}\n"
-    return signature.format(open=symbol[0], close=symbol[1], name=name, date=date)
+
+    signature_one = "{open}\n**\n * author :   \t{name}\n * created : \t{date}\n**\n{close}\n"
+
+    signature_mul = "{open}\n{open}**\n{open} * author :   \t{name}\n{open} * created : \t{date}\n{open}**\n{close}\n"
+
+    if symbol[0] in ['#']: #we can add other into a list and check if it is in it...
```
Comment (Harjacober) [inline signer.py]: If the symbol.py file has been updated as suggested, then this line should be replaced with `if symbol[2]:` (True for multi-line comment)

Comment (jeafreezy) [reply signer.py]: Yeah, thanks. I just made the changes as discussed.

**Comment by jeafreezy** [comment]
Code updated as discussed

**Pull request merged**